lib/drivers/medtronicDriver.js

```diff
@@ -0,0 +1,746 @@
+            }
+            callback(null, 'open');
+          });
+        },
+        model : function(callback){
+          getOneRecord(readModel(), false, function(err, result) {
+            if(err) {
+              return cb(err,null);
+            }
+            callback(null, {settings: {modelNumber: result.model}});
+          });
+        }
+      },
+      function(err, results){
+          progress(100);
+
+          if(!err){
+              data.connect = true;
+              _.assign(data, results.model);
+              return cb(null, data);
+          } else {
+              return cb(err,results);
+          }
+      });
+    },
+
+    fetchData: function (progress, data, cb) {
+      if(!isBrowser) {
+        data.fetchData = true;
+        return cb(null, data);
+      }
+      debug('in fetchData', data);
+
+      progress(0);
+      async.series({
+        readHistory : function(callback){
+          var count = 0;
+          var pages = [];
+
+          async.whilst(
+              function () { return count < 9; },
+              function (callback) {
+                  getOneRecord(sendCommand(MESSAGES.READ_HISTORY), true, function (err, result) {
+                    if(err) {
+                      return callback(err,null);
+                    }
+                    if(result) {
+                      getRecords(readPage(MESSAGES.READ_HISTORY,[0x01,count]), 4, function (err, result) {
+                        if(err) {
+                          return callback(err,null);
+                        }
+                        if(result) {
+                          pages[count] = result;
+                          count++;
+                          return callback(null, count);
+                        } else {
+                          return cb(new Error('No history'));
+                        }
+                      });
+                    } else {
+                      return cb(new Error('No history'));
+                    }
+                  });
+              },
+              function (err, n) {
+                  if(err) {
+                    return cb(err,null);
+                  } else {
+                    console.log('Read', n, 'pages');
+                    _.assign(data, { pages : pages });
+                    return cb(null, data);
+                  }
+              }
+          );
+        },
+      },
+      function(err, results){
+          if(!err){
+            progress(100);
+            data.fetchData = true;
+            cb(null, data);
+          } else {
+            return cb(err,results);
+          }
+      });
+    },
+
+    processData: function (progress, data, cb) {
+      debug('in processData');
+      data.settings.deviceId = data.settings.modelNumber + '-' + serial;
+      cfg.builder.setDefaults({ deviceId: data.settings.deviceId});
+      // TODO: get all time changes and most recent event
+      cfg.tzoUtil = new TZOUtil(cfg.timezone, new Date().toISOString(), []);
```

Comment:
Just as a note for the future, I _think_ you'll need two instances of `tzoUtil` because CGM data stream and pump data stream need to be bootstrapped separately. Just mentioning now as a seed to plant for code organization/architecture, perhaps.